pageserver/src/basebackup.rs

```diff
@@ -252,6 +255,30 @@ where
     async fn send_tarball(mut self) -> Result<(), BasebackupError> {
         // TODO include checksum
 
+        let normal_shutdown = if let Ok(checkpoint_bytes) =
```

Comment:
This desperately needs comments

Reply:
+1 on comments, and should probably be encapsulated into a member function on CheckPoint rather than inlined here.